src/lib/autoinstall/y2erb.rb

```diff
@@ -12,6 +13,12 @@ def self.render(path)
     class TemplateEnvironment
       include Yast::Logger
 
+      # @see {EFIDetector}
+      # @return [Boolean] whether the system is booted using EFI or not
+      def boot_efi?
+        (@efi_detector ||= EFIDetector.new).boot_efi?
```

Comment:
If you prefer, you could use a class method to cache the value (as it is not going to change).

```ruby
class EFIDetector
  def self.boot_efi?
    @detector ||= EFIDetector.new
    @detector.boot_efi?
  end
end
```

Reply:
I will move the method completely to a class method by now and will not cache anything at all.